### DISCUSSION TITLE
How to set the temperature of a wall

### DISCUSSION THREAD

**Discussion by RachelMeng0096** [AMReX-Combustion/PeleC#736] [Q&A]
Greetings everyone, I followed the manual to establish an isothermal wall, yet the computational results still indicate it's an adiabatic surface. Below are some of my settings; I would greatly appreciate any advice!
![image](https://github.com/AMReX-Combustion/PeleC/assets/153271449/6b65ae46-d120-44b9-9225-4e1d7b3c4632)

![image](https://github.com/AMReX-Combustion/PeleC/assets/153271449/b661a7b7-d479-444a-80f0-243503866959)

**Comment by baperry2** [comment]
It's difficult to tell from this image, but if I'm seeing things correctly I believe the walls are being treated correctly as isothermal. It appears that the temperature in the first cell adjacent to the isothermal wall has a slightly different temperature than the cells further from the wall.

Note that the state variables in the plot files are cell-centered quantities interpreted as representing integrals over the cell volume. Therefore, even when the heat fluxes from the isothermal wall are appropriately computed, it takes time to affect the cells adjacent to the wall, and, depending on what is happening in the interior of the domain, the temperature in the first cell may never exactly match the temperature you are specifying for the wall.

**Reply by Pockjob** [reply]
Hi @baperry2 , is it possibile to set the default noslipwall as isothermal? I use pelec.do_isothermal_walls in input file to do that but it does not work.

**Reply by baperry2** [reply]
Changing the default behavior would break existing cases that utilize the default, and there wouldn't be an obvious choice for a default isothermal temperature, so we can't make that change. 

For isothermal walls, ensure you are following the instructions [here](https://amrex-combustion.github.io/PeleC/BoundaryConditions.html#isothermal-walls) and take a look at [this case](https://github.com/AMReX-Combustion/PeleC/blob/development/Exec/RegTests/MassCons/masscons-isothermal.inp) for an example. Then if you are still having issues, you'll need to provide more information about your case in order for us to give useful debugging help.

**Reply by Pockjob** [reply]
Hi @baperry2, I used `pelec.eb_boundary_T=300.` and `pelec.eb_isothermal=1.`  in input file to set a EB as isothermal wall. But the result showed an adiabatic EB wall and no low-temperature boundary layer occurs. How can I make the EB an isothermal wall and create a boundary layer with a fixed temperature?
![T](https://github.com/AMReX-Combustion/PeleC/assets/154661154/5db738b1-1642-4368-b3e1-2e55e586be48)

**Reply by baperry2** [reply]
Those inputs look correct. Make sure you also have `pelec.diffuse_temperature = 1` and `pelec.diffuse_enth = 1`. It looks like you're simulating a high speed flow, perhaps the thermal boundary leyer is just too thin to be resolved. I'd recommend moving to a simpler flow configuration to verify the isothermal EB capability is working as you expect. For example, if you run your case, but quiescent, can you see the effect of the isothermal wall? You can also check out the `EB-C5` case for a minimal example with isothermal walls. 

As an additional note, if that is the geometry you are trying to simulate, you'd be better off just aligning the upper domain boundary with the wall and not using EB.
